src/Tokenizer/Token.php

```diff
@@ -166,11 +166,11 @@ public function equals($other, $caseSensitive = true)
             return $this->content === $otherPrototype;
         }
 
-        if (array_key_exists(0, $otherPrototype) && $this->id !== $otherPrototype[0]) {
+        if ($this->id !== $otherPrototype[0]) {
             return false;
         }
 
-        if (array_key_exists(1, $otherPrototype)) {
```

Comment:
I believe we can drop array_key_exists, one condition (regardless which one) is enough